sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -48,6 +51,24 @@ export interface ConnectionContext extends ConnectionContextBase {
    * the underlying amqp connection for the EventHub Client.
    */
   managementSession?: ManagementClient;
+
+  /**
+   * Indicates whether the connection is in the process of closing.
+   */
+  isConnectionClosing(): boolean;
+  /**
+   * Resolves once the connectionContext is ready to be opened.
```

Comment:
Its not really the connection context that gets opened... its the connection. Consider renaming `readyToOpen` to `readyToOpenConnection`?

On the other hand, the usage seems to be to wait until it is safe to initialize sender, receiver and management links. The connection is "opened" only if it wasnt already opened. So maybe name it to `readyToOpenLink`? We are ready to open links if
- connection is already open and is well
- connection is closed/disconnected
- but not if connection is in the process of closing or disconnecting

Reply:
Updated to readyToOpenLink and added more comments. Thanks, I do like this name better!